lib/services/account-api-service.js

```diff
@@ -189,5 +197,108 @@ function accountApiServiceFactory(
         });
     };
 
+    AccountApiService.prototype.createRole = function (newrole) {
+        var self = this;
+        var acl = this.acl;
+        return Promise.try(function () {
+            assert.string(newrole.role);
+            return waterline.roles.findOne({ role: newrole.role });
+        })
+        .then(function (entry) {
+            if (entry) {
+                // TODO: Replace with HttpError after errors are refactored
+                var err = new Errors.BaseError('role already exists');
+                err.status = 409;
+                throw err;
+            }
+            return waterline.roles.create({
+                role: newrole.role,
+                privileges: newrole.privileges
+            });
+        })
+        .then(function (entry) {
+            return acl.addRoleParents(entry.role, entry.privileges);
+        })
+        .then(function () {
+            return self.getRoleByName(newrole.role);
+        });
+    };
+
+    AccountApiService.prototype.getRoleByName = function (name) {
+        var self = this;
+        return Promise.try(function () {
+            assert.string(name);
+            return waterline.roles.findOne({ role: name });
+        })
+        .then(function (roleFound) {
+            if (!roleFound) {
+                throw new Errors.NotFoundError();
```

Comment:
We could use `waterline.roles.needByIdentifier` and remove the `!roleFound` condition altogether.  I'm fine with this either way though.